cogs/utils/generic.py

```diff
@@ -60,9 +65,29 @@ async def get_channels(self, ctx):
         await ctx.channel.send(f"```{output}```")
 
 
-    # Do stuff when a message is deleted
-    @commands.Cog.listener()
-    async def on_message_delete(self, message):
-        pass
+    # Send a message to a user on their birth day
+    @tasks.loop(hours=24)
+    async def birthday_alert(self):
+        # List to save id of members who already recieved a msg
+        member_ids_sent = []
+        guild_ids = await get_all_guilds()
+        for guild_id in guild_ids:
+            members = await get_all_members(guild_id)
+            for member in members:
+                member_id = member['member_id']
+                if member['birthday'] == None or member_id in member_ids_sent:
+                    continue
+                bday = datetime.strptime(member['birthday'], '%Y-%m-%d')
+                now = date.today()
+                if (bday.month, bday.day) == (now.month, now.day):
+                    user = self.bot.get_user(member_id)
+                    reply = f"Happy birthday, **{user.name}**!"
+                    await user.send(content=reply)
+                    member_ids_sent.append(member_id)
 
-# add blurbs, reminders, birthday alerts
\ No newline at end of file
+    # Wait until the bot is ready before the birthday_alert loop starts
+    @birthday_alert.before_loop
+    async def before_birthday_alert(self):
+        await self.bot.wait_until_ready()
```

Comment:
Absolutely love that you added a check here to make sure the bot is ready before starting the loop, great catch.